Lib/http/server.py

```diff
@@ -902,6 +937,24 @@ def guess_type(self, path):
             return guess
         return 'application/octet-stream'
 
+    def get_range(self):
+        """Return a tuple of (start, end) representing the range header in
+        the HTTP request. If the range header is missing or not resolvable,
+        None is returned. This only supports single part ranges.
+
+        """
+        range_header = self.headers.get('range')
+        if not range_header:
+            return None
+        m = re.match(r'bytes=(\d+)-(\d*)$', range_header)
+        if not m:
+            return None
+        start = m.group(1)
+        if not m.group(2):
+            return int(start), None
+        end = m.group(2)
+        return int(start), int(end)
```

Comment:
```suggestion
        start, end = int(start), int(end)
        if end < start:
        	return None
       	return start, end
```

According to specs:

>  A byte-content-range-spec with a byte-range-resp-spec whose last-byte-pos value is less than its first-byte-pos value, or whose instance-length value is less than or equal to its last-byte-pos value, is invalid. The recipient of an invalid byte-content-range-spec MUST ignore it and any content transferred along with it. 

Technically, we should just ignore the rest of the content but I'm not sure if this is what you want to do in general.

PS: I only suggested something based on the current regex but my suggestion should be updated if you also assume the presence of the instance length.

Reply:
Thanks for the heads up, I did forget to take into account the case where start is larger than end, I've fixed it and added a test for that.

Reply:
EDIT: Since I misread the section (I was on the response section and not on the request section), it appears that the syntax "bytes=-500" for the last 500 bytes is also valid. So maybe you want to change that as well?

See: https://datatracker.ietf.org/doc/id/draft-ietf-httpbis-p5-range-09.html#byte.ranges

Reply:
I noticed this part, the reason I didn't implement it at first was that it seemed like this usage wasn't very common, I referred to [the documentation on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range) and implemented the first two in the order it was provided. (I've tested Chrome's video progress bar, IDM, aria2, the mainstay tools, and they all seem to work fine.)

I wouldn't mind implementing suffix though, let me implement it now.

Reply:
Actually, I personally won't use it but I think it might be nice for people that want to test it and for those that expect the specs to be followed.

Reply:
Have implemented suffix in the latest commit.